src/index.ts

```diff
@@ -1,63 +1,69 @@
-import { token, botAdmins, prefixes } from './env';
-import CookiecordClient from 'cookiecord';
-import { Intents } from 'discord.js';
+import { Client, GatewayIntentBits, Partials } from 'discord.js';
+import { Bot } from './bot';
 import { getDB } from './db';
+import { token } from './env';
 import { hookLog } from './log';
 
-import { AutoroleModule } from './modules/autorole';
-import { EtcModule } from './modules/etc';
-import { HelpThreadModule } from './modules/helpthread';
-import { PlaygroundModule } from './modules/playground';
-import { RepModule } from './modules/rep';
-import { TwoslashModule } from './modules/twoslash';
-import { HelpModule } from './modules/help';
-import { SnippetModule } from './modules/snippet';
-import { HandbookModule } from './modules/handbook';
-import { ModModule } from './modules/mod';
+import { autoroleModule } from './modules/autorole';
+import { etcModule } from './modules/etc';
+import { handbookModule } from './modules/handbook';
+import { helpModule } from './modules/help';
+import { modModule } from './modules/mod';
+import { playgroundModule } from './modules/playground';
+import { repModule } from './modules/rep';
+import { twoslashModule } from './modules/twoslash';
+import { snippetModule } from './modules/snippet';
+import { helpThreadModule } from './modules/helpthread';
 
-const client = new CookiecordClient(
-	{
-		botAdmins,
-		prefix: prefixes,
+const client = new Client({
+	partials: [
+		Partials.Reaction,
+		Partials.Message,
+		Partials.User,
+		Partials.Channel,
+	],
+	allowedMentions: {
+		parse: ['users', 'roles'],
 	},
-	{
-		partials: ['REACTION', 'MESSAGE', 'USER', 'CHANNEL'],
-		allowedMentions: {
-			parse: ['users', 'roles'],
-		},
-		intents: new Intents([
-			'GUILDS',
-			'GUILD_MESSAGES',
-			'GUILD_MEMBERS',
-			'GUILD_MESSAGE_REACTIONS',
-			'DIRECT_MESSAGES',
-		]),
-	},
-).setMaxListeners(Infinity);
-
-for (const mod of [
-	AutoroleModule,
-	EtcModule,
-	HelpThreadModule,
-	PlaygroundModule,
-	RepModule,
-	TwoslashModule,
-	HelpModule,
-	SnippetModule,
-	HandbookModule,
-	ModModule,
-]) {
-	client.registerModule(mod);
-}
+	intents: [
+		GatewayIntentBits.Guilds,
+		GatewayIntentBits.GuildMessages,
+		GatewayIntentBits.GuildMembers,
+		GatewayIntentBits.GuildMessageReactions,
+		GatewayIntentBits.DirectMessages,
+		GatewayIntentBits.MessageContent,
+	],
+}).setMaxListeners(Infinity);
 
-getDB(); // prepare the db for later
+getDB().then(() => client.login(token));
 
-client.login(token);
```

Comment:
Why serially? `Promise.all`? You have to await the `getDB` promise every time afterwards too if too much hasn't changed, and doing it here was just so the first command wouldn't take too long if the bot had time to rest.

Reply:
We only ever call `getDB` once, the entities implicitly use the data source - I do it serially so that we only init modules which try to use the DB once we are sure we actually have it. With the previous code, it was possible to boot up the bot without actually having a DB, and then getting promise rejections when trying to use it.

Reply:
Oh, huh. I guess it changed - the original idea I had was it could start a bit faster and work without DB for the modules that didn't *need* it since I think it used to go through the `getDB` handle for the entities.